src/ocean_emulators/utils/data.py

```diff
@@ -471,3 +502,28 @@ def unnormalize_tensor_prognostic(
         unnorm = torch.where(self.wet_mask.to(data.device) == 0, fill_value, unnorm)
         unnorm = unnorm.to(data.dtype)
         return unnorm
+
+    def unnormalize_tensor_boundary(
+        self, data: torch.Tensor, fill_value=float("nan")
+    ) -> torch.Tensor:
+        """Unnormalize boundary tensor."""
+        tensor_mean = self._to_tensor(self._boundary_mean_np, data.device)
+        tensor_std = self._to_tensor(self._boundary_std_np, data.device)
+
+        if data.ndim == 4:
+            assert data.shape[1] == self._boundary_mean_np.shape[0]
+            tensor_mean = tensor_mean.reshape([1, -1, 1, 1])
+            tensor_std = tensor_std.reshape([1, -1, 1, 1])
+        elif data.ndim == 5:
+            assert data.shape[2] == self._boundary_mean_np.shape[0]
+            tensor_mean = tensor_mean.reshape([1, 1, -1, 1, 1])
+            tensor_std = tensor_std.reshape([1, 1, -1, 1, 1])
+        else:
+            raise ValueError(f"Invalid data shape: {data.shape}")
```

Comment:
Do you think something like this would work instead?

```suggestion
        expand_var_dim = [1] * data.ndim
        expand_var_dim[-3] = -1
        assert data.shape[-3] == self._boundary_mean_np.shape[0]
        tensor_mean = tensor_mean.reshape(expand_var_dim)
        tensor_std = tensor_std.reshape(expand_var_dim)
```